Better handling of current project edge cases

Twixes opened the pull request:
## Changes

Resolves #6896 and a few other problems:
- lemonade UI looking awkward when there's no current project and/or when project creation is unavailable for the user
- user that lost their access to the current project not being redirected away
- org/project creation modals not being helpful

## How did you test this code?

Manually by bringing the edge cases described above into the instance.

Twixes committed:
Fix handling of the user being shut out of the current project

Twixes committed:
Fix blank screen when there's no current project

Twixes committed:
Properly handle current project state edge cases in UI refresh

Twixes committed:
Add more information to org/project creation modals

Twixes committed:
Update ErrorProjectUnavailable.tsx

mariusandra commented on frontend/src/scenes/App.tsx:
```diff
@@ -67,15 +66,14 @@ export function App(): JSX.Element | null {
     const { showApp, showingDelayedSpinner } = useValues(appLogic)
     const { user } = useValues(userLogic)
     const { currentTeamId } = useValues(teamLogic)
-    const { sceneConfig } = useValues(sceneLogic({ scenes: appScenes }))
```
If you don't need the var here, can you replace this with `useMountedLogic(sceneLogic({ scenes }))` instead, as the `<LoadedSceneLogics />` below also uses sceneLogic, before AppScene. It'd be an endless game of cat and mouse, so just `useMountedValues` at the highest level will save the day.

mariusandra reviewed (changes requested):
One comment, and there's an infinite loop in jest.

Twixes committed:
Use feedback

Twixes committed:
Fix API mocking for Jest

Twixes committed:
Update organizationLogic.test.ts

Twixes commented:
Fixed. Also added some console logging – it generally doesn't make sense in production, but in this case it feels better to have this hint for various debugging purposes (in logic tests but also to know where the redirects are coming from in prod) – though I don't feel strongly about this.

Twixes committed:
Merge branch 'master' into private-project-handling
